migrations/state_test.go

```
@@ -0,0 +1,264 @@
+
+	expectedState := &State{
+		Applied: []string{"migration1", "migration2"},
+		LastRun: time.Now(),
+	}
+
+	data, err := json.MarshalIndent(expectedState, "", "  ")
+	if err != nil {
+		t.Fatalf("failed to marshal state: %v", err)
+	}
+
+	err = os.WriteFile(statePath, data, 0644)
```

golangci-lint (2.5.0)

[medium] G306: Expect WriteFile permissions to be 0600 or less

(gosec)